src/main/java/org/itsallcode/openfasttrace/importer/tag/TagImporter.java

```diff
@@ -22,90 +22,54 @@
  * #L%
  */
 
-import java.io.BufferedReader;
-import java.io.IOException;
-import java.util.logging.Logger;
-import java.util.regex.Matcher;
-import java.util.regex.Pattern;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
 
-import org.itsallcode.openfasttrace.core.SpecificationItemId;
-import org.itsallcode.openfasttrace.importer.*;
+import org.itsallcode.openfasttrace.importer.ImportEventListener;
+import org.itsallcode.openfasttrace.importer.Importer;
+import org.itsallcode.openfasttrace.importer.LineReader;
+import org.itsallcode.openfasttrace.importer.LineReader.LineConsumer;
 import org.itsallcode.openfasttrace.importer.input.InputFile;
+import org.itsallcode.openfasttrace.importer.tag.config.PathConfig;
 
 /**
  * {@link Importer} for tags in source code files.
  */
 // [impl->dsn~import.full-coverage-tag~1]
 class TagImporter implements Importer
 {
-    private static final Logger LOG = Logger.getLogger(TagImporter.class.getName());
-    private static final Pattern COVERED_ID_PATTERN = SpecificationItemId.ID_PATTERN;
-    private static final String COVERING_ARTIFACT_TYPE_PATTERN = "\\p{Alpha}+";
-    private static final String TAG_PREFIX_PATTERN = "\\[";
-    private static final String TAG_SUFFIX_PATTERN = "\\]";
-
+    private final LineConsumer lineImporter;
     private final InputFile file;
-    private final ImportEventListener listener;
-    private final Pattern tagPattern;
 
-    public TagImporter(final InputFile file, final ImportEventListener listener)
+    public TagImporter(final Optional<PathConfig> config, final InputFile file,
+            final ImportEventListener listener)
     {
-        this.file = file;
-        this.listener = listener;
-        this.tagPattern = Pattern.compile(TAG_PREFIX_PATTERN //
-                + "(" + COVERING_ARTIFACT_TYPE_PATTERN + ")" //
-                + "->" //
-                + "(" + COVERED_ID_PATTERN + ")" //
-                + TAG_SUFFIX_PATTERN);
+        this(createLineImporter(config, file, listener), file);
```

Comment:
Hard to read. Please user intermediate variable "lineConsumer".

Also please use Consumer and Importer consistently.